### PR TITLE
PLUGIN-899 fix custom formats

### PR DESCRIPTION
**Pull request by albertshau**
Fixed a bug where custom formats would be disallowed during
plugin validation.

### CODE REVIEW

**Review by CuriousVini — APPROVED** [review]

File: src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java [hunk 1]
```diff
@@ -493,8 +485,8 @@ public String getPath() {
     }
 
     @Override
-    public FileFormat getFormat() {
-      return FileFormat.from(format, FileFormat::canWrite);
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java:497]: you are dropping `::canWrite` here. Is it intended?

File: src/main/java/io/cdap/plugin/gcp/gcs/Formats.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * Utility class for GCS formats.
+ */
+public class Formats {
+
+  private Formats() {
+    // no-op
+  }
+
+  /**
+   * Get the plugin name for the given format. Older versions of the source and sink allowed case insensitive
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/gcs/Formats.java:31]: If it returns plugin name, should it return "format-avro"? It's the artifact name I believe

File: src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java [hunk 3]
```diff
@@ -493,8 +485,8 @@ public String getPath() {
...
+    public String getFormatName() {
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java:488]: How is it going to be used?

File: src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java (hunk 1 above)
Comment (albertshau) [reply src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java:497]: yes, this method is unused code.

File: src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java (hunk 3 above)
Comment (albertshau) [reply src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java:488]: this gets used by https://github.com/cdapio/hydrator-plugins/blob/19a6d06a538714a5fdcde15eac9084b5bfe2fbba/format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSink.java#L90, it just asks CDAP for a plugin with this name

File: src/main/java/io/cdap/plugin/gcp/gcs/Formats.java (hunk 2 above)
Comment (albertshau) [reply src/main/java/io/cdap/plugin/gcp/gcs/Formats.java:31]: we want the plugin name and not the artifact name, which is what gets passed to CDAP's usePlugin() API call at https://github.com/cdapio/hydrator-plugins/blob/19a6d06a538714a5fdcde15eac9084b5bfe2fbba/format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSource.java#L124

**Review by tivv — APPROVED** [review]

**Pull request merged**